packages/router/router.ts

```diff
@@ -2627,22 +2628,19 @@ async function callLoaderOrAction(
         "Redirects returned/thrown from loaders/actions must have a Location header"
       );
 
-      // Check if this an external redirect that goes to a new origin
-      let currentUrl = new URL(request.url);
-      let currentOrigin = currentUrl.origin;
-      let newOrigin = new URL(location, currentOrigin).origin;
-      let external = newOrigin !== currentOrigin;
+      let isAbsolute =
+        /^[a-z+]+:\/\//i.test(location) || location.startsWith("//");
```

Comment:
Instead we'll just look for the presence of a leading protocol or a protocol-less URL to determine if we can skip relative routing logic